enterprise/server/remote_execution/workspace/workspace.go

```diff
@@ -139,6 +144,38 @@ func (ws *Workspace) DownloadInputs(ctx context.Context, tree *repb.Tree) (*dirt
 	return txInfo, err
 }
 
+func (ws *Workspace) CleanInputsIfNecessary(keep map[string]*repb.Digest) error {
+	if ws.opts.CleanInputs == "" {
+		return nil
+	}
+	inputFilesToCleanUp := make(map[string]*repb.Digest)
+	for path, digest := range ws.Inputs {
+		if ws.opts.CleanInputs == "*" {
+			inputFilesToCleanUp[path] = digest
+			continue
+		}
+		patterns := strings.Split(ws.opts.CleanInputs, ",")
+		for _, pattern := range patterns {
+			if strings.Contains(path, pattern) {
+				inputFilesToCleanUp[path] = digest
+				continue
+			}
+		}
```

Comment:
I think we should use `filepath.Match`on the basename of the path (`filepath.Base`), instead of supporting a mix of `strings.Contains` and an explicit "match-all" pattern here:

* It supports single-star globs already, so there's no need to treat `*` specially (which might mislead people into thinking that other globs will work as well).
* It's less error-prone since it matches the whole string. For example `contains(".h")` will also match `foo.hs` while `match("*.h")` will only match `.h` files.
* It's backwards-compatible with full glob expressions (i.e. globs including `**`, like `include/**/*.h`), if we want to support that later on

Glob patterns would need an explicit `*`, like `*.h,*.framework`, and the impl would have to work on the basename and not the full path. https://play.golang.org/p/FiX_WYiS-c1

Reply:
Resolved offline - `filepath.Match` + `filepath.Base` won't work if the patterns are trying to match a directory rather than a basename. We can later use a separate platform prop like `clean-workspace-input-patterns` if we want full glob support

Reply:
Ended up switching to a glob that supports ** since that's what my original intent was - and we're already pulling in a library that supports this.